src/material/button-toggle/button-toggle.ts

```diff
@@ -257,6 +262,41 @@ export class MatButtonToggleGroup implements ControlValueAccessor, OnInit, After
     this.disabled = isDisabled;
   }
 
+  /** Handle keydown event calling to single-select button toggle. */
+  protected _keydown(event: KeyboardEvent) {
+    if (this.multiple || this.disabled) {
+      return;
+    }
+
+    const target = event.target as HTMLButtonElement;
+    const buttonId = target.id;
+    const index = this._buttonToggles.toArray().findIndex(toggle => {
+      return toggle.buttonId === buttonId;
+    });
+
+    let nextButton;
+    switch (event.keyCode) {
+      case SPACE:
+      case ENTER:
+        nextButton = this._buttonToggles.get(index);
+        break;
+      case UP_ARROW:
+      case LEFT_ARROW:
```

Comment:
Do we need to account for RTL layouts here? I don't see any logic for it in `_getNextIndex`.

Reply:
I didn’t consider how people who use RTL keyboards would interact with this. The reason for that is I want the single-select toggle button to work just like the Mat radio group keyboard-wise. I looked into the Mat radio group, but I couldn't find any specific keyboard interactions for RTL. So, I'm guessing that it's not supported.

Reply:
Ah I mean the Mat radio group does not have RTL logic and single toggle button group is a radio group under the hood. But if we want to implement this, do we create a new variable (say direction) taking RTL/LTR as an input of toggle button group component and handle them differently keyboard-wise?

Reply:
There's a DI token you can use to figure out the current direction: https://github.com/angular/components/blob/main/src/material/menu/menu-trigger.ts#L265

Then you can use it to invert the left/right arrow: https://github.com/angular/components/blob/main/src/material/menu/menu-trigger.ts#L627

Reply:
Added RTL support.